src/kudu/consensus/quorum_util.cc

```diff
@@ -817,5 +838,32 @@ bool ShouldEvictReplica(const RaftConfigPB& config,
   return should_evict;
 }
 
+void GetRegionalCountsFromConfig(
+    const RaftConfigPB& config, const std::string& leader_uuid,
+    std::map<std::string, int>* regional_count, std::string* leader_region) {
+  CHECK(regional_count);
+  CHECK(leader_uuid.empty() || leader_region != nullptr);
+  std::set<std::string> unique_uuids;
+  for (const RaftPeerPB& peer : config.peers()) {
+    // Only consider peers that are voters.
+    if (!peer.has_member_type() ||
+        peer.member_type() != RaftPeerPB::VOTER) {
+      continue;
+    }
+    CHECK(peer.has_permanent_uuid() && peer.has_attrs() &&
```

Comment:
better to skip the peer and report error message than crashing server here.

Reply:
Yeah, if upstream wants a flexiraft implementation, we would need a mechanism to define subsets of servers -- we can think about generalizing choosing subsets once our use case is done.